packages/neon-domain/src/provider/NeoNS/core.ts

```diff
@@ -0,0 +1,49 @@
+import { logging, u, sc, wallet } from "@cityofzion/neon-core";
+import {
+   resolve,
+} from "../common";
+
+const log = logging.default("neon-domain");
+
+const operation = "resolve";
+
+/**
+ * Get balances of NEO and GAS for an address
+ * @param url - URL of a neonDB service.
+ * @param address - Address to check.
+ * @return  Balance of address
+ */
+export async function resolveDomain(
+  url: string,
+  contract: string,
+  domain: string,
+  tld: string

```

Comment:
The `domain` and `tld` args both seem like unnecessary duplicate pieces of data.  They are both things that can be derived from `url`.